src/Misc/layoutbin/powershell/Use-VstsTaskScript.ps1

```diff
@@ -0,0 +1,42 @@
+<# 
+A PowerShell script that is used to invoke a VSTS task script. This script is used by the VSTS task runner to invoke the task script.
+This script resplaces some legacy stuff in PowerShell3Handler.cs and turns it into a dedicated function. 
+since it is parameterized it can be signed and trusted for WDAC and CLM.
+This is the original code that has been repalced - 
+@". ([scriptblock]::Create('if ([Console]::InputEncoding -is [Text.UTF8Encoding] -and [Console]::InputEncoding.GetPreamble().Length -ne 0) {{ [Console]::InputEncoding = New-Object Text.UTF8Encoding $false }} if (!$PSHOME) {{ $null = Get-Item -LiteralPath ''variable:PSHOME'' }} else {{ Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'')) ; Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1'')) }}')) 2>&1 | ForEach-Object {{ Write-Verbose $_.Exception.Message -Verbose }} ; Import-Module -Name '{0}' -ArgumentList @{{ NonInteractive = $true }} -ErrorAction Stop ; $VerbosePreference = '{1}' ; $DebugPreference = '{1}' ; Invoke-VstsTaskScript -ScriptBlock ([scriptblock]::Create('. ''{2}'''))",
+#>
+
+param ( 
+    [Parameter(mandatory = $true)]$name, # Argument 0 
+    [Parameter(mandatory = $true)]$DebugOption, # Argument 1
+    [Parameter(mandatory = $true)]$scriptBlockString	# Argument 2 
+)
+
+if ([Console]::InputEncoding -is [Text.UTF8Encoding] -and [Console]::InputEncoding.GetPreamble().Length -ne 0) {
+    [Console]::InputEncoding = New-Object Text.UTF8Encoding $false 
+} 
+
+if (!$PSHOME) { 
+    $null = Get-Item -LiteralPath 'variable:PSHOME' 
+}
+else { 
+    Import-Module -Name ([System.IO.Path]::Combine($PSHOME, 'Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1')) 
+    Import-Module -Name ([System.IO.Path]::Combine($PSHOME, 'Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1')) 
+}
```

Comment:
[NotBlocking]: What does this mean actually? If $PSHome is not defined then try to get it and skip import modules?
Can't find any sense here but I see you copied that from the original script string so I would just to add `# TODO` for us to check that later.
Or if you have thoughts why this was added, please share with me🙂
Feel free to resolve.

Reply:
Unknown as to the case, but not my logic. I've never seen pshome being left undefined. I copied the original logic over. I've added the todo mention to check at a later point.